### ISSUE TITLE
is there a way to suppress tokens during generation?

### ISSUE THREAD

**Issue by yangkky** [ml-jku/Prot-xLSTM#1]
HF has the following functionality: 

```
from transformers import SuppressTokensLogitsProcessor

sup = SuppressTokensLogitsProcessor([t for t in all_tokens if not t in allowed_tokens], device=DEVICE)
generated = model.module.generate(start, do_sample=True, logits_processor=[sup],
                                                             temperature=args.temp, min_p=args.min_p, num_beams=1, max_new_tokens=max_len,
                                                      use_cache=True)
```

Is it possible to do something like this to prevent generation of mask tokens?
